substrate/frame/election-provider-multi-block/src/signed/tests.rs

```diff
@@ -224,16 +231,364 @@ mod calls {
+				NotAcceptingSubmissions::<Runtime>,
+			);
+		})
+	}
+
+	#[test]
+	fn register_and_submit_page_and_bail_prohibitted_in_phase_emergency() {
+		ExtBuilder::default().build_and_execute(|| {
+			let account_id = 99;
+			compute_snapshot_checked();
+			roll_to_phase(Phase::Emergency);
+
+			assert_err!(
+				SignedPallet::register(RuntimeOrigin::signed(account_id), Default::default()),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+
+			assert_err!(
+				SignedPallet::submit_page(
+					RuntimeOrigin::signed(account_id),
+					0,
+					Some(Default::default())
+				),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+
+			assert_err!(
+				SignedPallet::bail(RuntimeOrigin::signed(account_id)),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+		})
+	}
+
+	#[test]
+	fn bail_while_having_no_submissions_does_not_modify_balances() {
+		ExtBuilder::default().build_and_execute(|| {
+			roll_to_phase(Phase::Signed);
+
+			// expected base deposit with 0 submissions in the queue.
+			let base_deposit = <Runtime as Config>::DepositBase::convert(0);
+			let page_deposit = <Runtime as Config>::DepositPerPage::get();
+			assert!(base_deposit != 0 && page_deposit != 0 && base_deposit != page_deposit);
+
+			let account_id = 99;
+
+			// account_id has 100 free balance and 0 held balance for elections.
+			// As configured in
+			// pallet-epm-mb/substrate/frame/election-provider-multi-block/src/mock/mod.rs ->
+			// pallet_balances::GenesisConfig::<T>
+			assert_eq!(balances(account_id), (100, 0));
+
+			assert_ok!(SignedPallet::register(
+				RuntimeOrigin::signed(account_id),
+				Default::default()
+			));
+
+			// free balance and held deposit updated as expected.
+			assert_eq!(balances(account_id), (100 - base_deposit, base_deposit));
+
+			// submit page
+			assert_ok!(SignedPallet::submit_page(
+				RuntimeOrigin::signed(account_id),
+				0,
+				Some(Default::default())
+			));
+
+			// free balance and held deposit updated as expected
+			assert_eq!(
+				balances(account_id),
+				(100 - base_deposit - page_deposit, base_deposit + page_deposit)
+			);
+
+			let bailing_account_id = 91;
+
+			// bailing_account_id has 100 free balance and 0 held balance for elections.
+			// As configured in
+			// pallet-epm-mb/substrate/frame/election-provider-multi-block/src/mock/mod.rs ->
+			// pallet_balances::GenesisConfig::<T>
+			assert_eq!(balances(bailing_account_id), (100, 0));
+
+			// account 1 submitted nothing, so bail should have no effect and return error
+			assert_err!(
+				SignedPallet::bail(RuntimeOrigin::signed(bailing_account_id)),
+				SubmissionNotRegistered::<Runtime>
+			);
+
+			// balance of account with a submission as before
+			assert_eq!(
+				balances(account_id),
+				(100 - base_deposit - page_deposit, base_deposit + page_deposit)
+			);
+
+			// bailing account balance unchanged
+			assert_eq!(balances(bailing_account_id), (100, 0));
+		})
+	}
+
+	#[test]
+	fn force_clear_submission_fails_if_called_by_account_none() {
```

Comment:
This is a test for force_clear but I guess you mean the one for bail. 
Anyway, if I understand correctly you would put all the test cases for "bail" function in one test function.
If yes, then I would be against it. It would surely provide the coverage we need and verify all we need but it would result in overloaded tests, checking lots of things, instead of many tests, each checking very specific outcome.
So let me understand your reasoning here.

Reply:
Quick 2-cents from me: in this case, testing different origins is a simple procedure, so I get @Zebedeusz's point of having different `#[test]`s

If testing different origins required a (more or less) similar setup everytime, doing it all in the same test would make more sense as @gpestana suggests.

Example: a verbose [integration test](https://github.com/paritytech/polkadot-sdk/pull/6377/files#diff-8501c51f73a211c9a61241de6aa68de2e8f87d086c523fd52c7cddcfedec4d59R79) for Westend's people parachain checking for different wrong origins.